Automattic/wp-calypso
# Components: Remove `FormToggle` in favor of Gutenberg's `ToggleControl`

#### Changes proposed in this Pull Request

In #52915 we're removing the last real need for a `FormToggle` component in Calypso. With that merged, `FormToggle` becomes a simple proxy of the `ToggleControl` component, and the only real difference is the way that the label is provided - in `FormToggle` we provide those as `children`, while in `ToggleControl` that's being done with the `label` prop. So `FormToggle` can just be removed in favor of `ToggleControl`, there is no real need to maintain that abstraction anymore.

This PR updates all of the `FormToggle` instances to use `ToggleControl`, and removes `FormToggle` altogether.

#### Testing instructions

* Verify all tests pass.
* Verify we correctly replace all `FormToggle` `children` props with `label` prop when using `ToggleControl`. 
* Go through some of the major areas that contain toggles and verify they work and look the same way as before:
  * `/settings/general/:site` where `:site` is a Jetpack site - go through all the settings tabs.
  * `/extensions/wp-super-cache/:site` where `:site` is a Jetpack site - go through all the settings tabs.
*  Go to `/devdocs/wordpress-components-gallery`  and verify the form toggle example is unaffected by this change.

#### Notes

~This PR is blocked by #52915 and needs to be rebased after it lands.~

There are pre-existing ESLint errors that are expected and are not being fixed in this PR to avoid inflating the PR.

We're also fixing a couple of minor spacing discrepancies and inline comments as we go.
